clients/js/main.ts

```diff
@@ -267,20 +277,44 @@ yargs(hideBin(process.argv))
       ) {
         throw Error(`Unknown network: ${network}`);
       }
+      let chain = argv["chain"]
       let module = argv["module"] as "Core" | "NFTBridge" | "TokenBridge";
+      let addr = ""
       switch (module) {
         case "Core":
-          console.log(CONTRACTS[network][argv["chain"]]["core"]);
+          addr = CONTRACTS[network][chain]["core"];
           break;
         case "NFTBridge":
-          console.log(CONTRACTS[network][argv["chain"]]["nft_bridge"]);
+          addr = CONTRACTS[network][chain]["nft_bridge"];
           break;
         case "TokenBridge":
-          console.log(CONTRACTS[network][argv["chain"]]["token_bridge"]);
+          addr = CONTRACTS[network][chain]["token_bridge"];
           break;
         default:
           impossible(module);
       }
+      if (argv["emitter"]) {
+        if (chain === "solana" || chain === "pythnet") { // TODO: Create an isSolanaChain()
+          addr = await getEmitterAddressSolana(addr)
+        } else if (isTerraChain(chain)) {
+          addr = await getEmitterAddressTerra(addr)
+        } else if (chain === "algorand") {
+          if (network !== "MAINNET") {
+            throw Error(`unable to look up algorand emitter address for ${network}`);
+          }
+          addr = "25e716e0618d9f38b603a97cc42db659069c0f5185230e5e61679fa876191ec4"
+        } else if (chain === "near") {
+          if (network !== "MAINNET") {
+            throw Error(`unable to look up near emitter address for ${network}`);
+          }
+          addr = "148410499d3fcda4dcfd68a1ebfcdddda16ab28326448d4aae4d2f0465cdfcb7"          
+        } else {
+          addr = getEmitterAddressEth(addr)
+        }
+
+        addr += " " + toChainId(chain)
```

Comment:
what's the use case for putting the chain id into the output? My worry is that it will make composability a little more cumbersome when used in shell scripts. For example a common use case for me is doing things like
```shell
cast call $(worm contract mainnet ethereum Core) "chainId()" --rpc-url $(worm rpc mainnet ethereum)
```
but with this change I would have to do
```shell
cast call $(worm contract mainnet ethereum Core | cut -d' ' -f1) "chainId()" --rpc-url $(worm rpc mainnet ethereum)
```

Reply:
My script needs both the emitter address and the chain ID and I got lazy...